Vialer/Helpers/VialerStats.swift

```diff
@@ -160,13 +163,32 @@ import Foundation
 
         defaultData[VialerStatsConstants.APIKeys.direction] = VialerStatsConstants.Direction.inbound
         defaultData[VialerStatsConstants.APIKeys.callSetupSuccessful] = "false"
-        defaultData[VialerStatsConstants.APIKeys.failedReason] = "INSUFFICIENT_NETWORK"
+        defaultData[VialerStatsConstants.APIKeys.failedReason] = VialerStatsConstants.FailedReason.insufficientNetwork
         defaultData[VialerStatsConstants.APIKeys.timeToInitialResponse] = String(timeToInitialReport)
         defaultData.removeValue(forKey: VialerStatsConstants.APIKeys.attempt)
         
         sendMetrics()
     }
-
+    
+    @objc func incomingCallFailedDeclined(call: VSLCall){
```

Comment:
I think the Declined / Originator_cancel / call completed elsewhere can be contained in a single function i think.

Reply:
Yes, I created the method on viali-3477 and after your review I will merge it to the epic branch then rebase from epic to this one and finish this ticket while resolving the conflicts. So to finish this one I wait for 3477 to pass the review.